src/main/java/org/verdictdb/jdbc41/VerdictStatement.java

```diff
@@ -45,7 +45,7 @@ public VerdictStatement(VerdictConnection jdbcConn, VerdictContext context) {
   @Override
   public boolean execute(String sql) throws SQLException {
     try {
-      result = executionContext.sql(sql);
+      result = context.sql(sql);
```

Comment:
I wonder if this is an intended behavior. My understanding was Connection:VerdictContext = Statement:ExecutionContext. If this was to handle bypass, I would suggest moving the logic into ExecutionContext, which still makes sense.

Reply:
It sounds good in general. But I still wonder how to close ExecutionContext when VerdictStatement.close() is called if VerdictStatement does not have the reference to ExecutionContext.

I think their conceptual relationship could be like, when a new VerdictStatement is created, VerdictConnection requests VerdictContext for a dedicated ExecutionContext, and then VerdictStatement simply uses the assigned ExecutionContext for query processing.

I agree that having a reference upper-level structure isn't good.

Reply:
Option 1: We can let VerdictContext to manage all of its ExecutionContext and have VerdictStatement to use its VerdictContext to close its associated ExecutionContext (e.g., using Map)

Option 2: We just have 1->2->4 and 3->4 execution path where every query processing logic goes into 4.

For now, I will use Option 2 as suggested while removing references to upper-level class/structures.